ray-operator/test/e2e/rayjob_suspend_test.go

```diff
@@ -93,4 +93,58 @@ func TestRayJobSuspend(t *testing.T) {
 		test.Expect(err).NotTo(HaveOccurred())
 		test.T().Logf("Deleted RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
 	})
+
+	test.T().Run("Create a suspended RayJob, and then resume it.", func(t *testing.T) {
+		// RayJob
+		rayJob := &rayv1.RayJob{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: rayv1.GroupVersion.String(),
+				Kind:       "RayJob",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "counter",
+				Namespace: namespace.Name,
+			},
+			Spec: rayv1.RayJobSpec{
+				Suspend:        true,
+				RayClusterSpec: newRayClusterSpec(mountConfigMap(jobs, "/home/ray/jobs")),
+				Entrypoint:     "python /home/ray/jobs/counter.py",
+				RuntimeEnvYAML: `
+env_vars:
+  counter_name: test_counter
+`,
+				ShutdownAfterJobFinishes: true,
+				SubmitterPodTemplate:     jobSubmitterPodTemplate(),
+			},
+		}
+		rayJob, err = test.Client().Ray().RayV1().RayJobs(namespace.Name).Create(test.Ctx(), rayJob, metav1.CreateOptions{})
+		test.Expect(err).NotTo(HaveOccurred())
+		test.T().Logf("Created RayJob %s/%s successfully", rayJob.Namespace, rayJob.Name)
+
+		test.T().Logf("Waiting for RayJob %s/%s to be 'Suspended'", rayJob.Namespace, rayJob.Name)
+		test.Eventually(RayJob(test, rayJob.Namespace, rayJob.Name), TestTimeoutMedium).
+			Should(WithTransform(RayJobDeploymentStatus, Equal(rayv1.JobDeploymentStatusSuspended)))
+
+		// Refresh the RayJob statuzs
+		rayJob = GetRayJob(test, rayJob.Namespace, rayJob.Name)
+
+		test.T().Logf("Resume the RayJob by updating `suspend` to false.")
+		rayJob.Spec.Suspend = false
+		// TODO (kevin85421): We may need to retry `Update` if 409 conflict makes the test flaky.
```

Comment:
I might use `patch` instead, as mentioned in [this discussion](https://github.com/ray-project/kuberay/pull/1783#discussion_r1441484684), after I acquire sufficient related knowledge (e.g., server-side apply).

Reply:
I've created #1811 that's related to server-side apply.

Reply:
Yes, I should be able to work on it soon.